frontend/src/metabase/questions/containers/MoveToCollection.jsx

```diff
@@ -93,3 +94,9 @@ export default class MoveToCollection extends Component {
         )
     }
 }
+
+MoveToCollection.defaultProps = {
+    setCollection: async (props, collection) => {
+        await props.dispatch(setCollection(props.questionId, collection.id, true))
```

Comment:
@tlrobinson Could use some guidance from you on how best to handle the async bits of this now that the call to the action is moved here.